easybuild/easyblocks/c/cuda.py

```diff
@@ -17,21 +17,42 @@
 @author: George Tsouloupas (Cyprus Institute)
 @author: Fotis Georgatos (Uni.lu)
 @author: Kenneth Hoste (Ghent University)
+@author: Damian Alvarez (Forschungszentrum Juelich)
 """
 import os
+import stat
+
 from distutils.version import LooseVersion
 
 from easybuild.easyblocks.generic.binary import Binary
-from easybuild.tools.filetools import patch_perl_script_autoflush
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.filetools import adjust_permissions, patch_perl_script_autoflush, write_file
 from easybuild.tools.run import run_cmd, run_cmd_qa
 from easybuild.tools.systemtools import get_shared_lib_ext
 
+# Wrapper script definition
+wrapper = """#!/bin/sh
+echo "$@" | grep -e '-ccbin' -e '--compiler-bindir' > /dev/null
+if [ $? -eq 0 ];
+then
+        echo "ERROR: do not set -ccbin or --compiler-bindir when using the `basename $0` wrapper"
+else
+        nvcc -ccbin=%s "$@"
+        exit $?
+fi """
 
 class EB_CUDA(Binary):
     """
     Support for installing CUDA.
     """
 
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'host_compilers': [[], "Host compilers for which a wrapper will be generated", CUSTOM]
```

Comment:
it's a good idea to avoid using a list as a default value

better use `None`, but you'll have to take it into account where you're using `self.cfg['host_compilers']`, for example with

```python
for comp in (self.cfg['host_compilers'] or []):
```